pkg/cloud/azure/actuators/machine/machine_status.go

```diff
@@ -0,0 +1,120 @@
+package machine
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+
+	"github.com/platform9/azure-provider/pkg/cloud/azure/services/resourcemanagement"
+	"k8s.io/apimachinery/pkg/runtime/serializer/json"
+	clusterv1 "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+	"sigs.k8s.io/cluster-api/pkg/util"
+)
+
+// holds the machine status under an annotation.
+// TODO: implement MachineStatus once the API is stable
+
+type MachineStatus *clusterv1.Machine
+type AnnotationKey string
+
+const (
+	Name           AnnotationKey = "azure-name"
+	ResourceGroup  AnnotationKey = "azure-rg"
+	InstanceStatus AnnotationKey = "instance-status"
+)
+
+func (azure *AzureClient) status(m *clusterv1.Machine) (MachineStatus, error) {
+	if azure.client == nil {
+		return nil, nil
+	}
+	currentMachine, err := util.GetMachineIfExists(azure.client, m.ObjectMeta.Namespace, m.ObjectMeta.Name)
+	if err != nil {
```

Comment:
I left the two checks to differentiate between the cases when the machine is actually because it has been deleted and just erroring on attempting to call the query the clients to check for machine existence.

Can we merge them in a way without losing that info?